vi/day20.md

```diff
@@ -0,0 +1,484 @@
+# Day 20: RxJS Creation
+
+In the previous chapter, we've learned about `Observable` and how to create one manually:
+
+```ts
+const observable = new Observable(function subscribe(observer) {
+  const id = setTimeout(() => {
+    observer.next('Hello Rxjs');
+    observer.complete();
+  }, 1000);
+  return function unsubscribe() {
+    clearTimeout(id);
+  };
+});
+```
+
+Some are probably wondering: "Do we have to remember this syntax to work with `Observable`? `next`, then `complete`, and also `unsubscribe`? This is a little too much of a hassle then". To answer this question, we will start exploring some **Operators** and **Functions** from **RxJS** that are used to create `Observable`.
+
+## Preface
+
+This is a common `Observer` that we'll be using throughout this post. If there is any example that uses a different `Observer`, I'll point it out.
+
+```ts
+const observer = {
+  next: (val) => console.log('next:', val),
+  error: (err) => console.log('error: err),
+  complete: () => console.log('complete'),
+};
+```
+
+## `of()`
+
+This is a function used to create an `Observable` from any type of value: primitives, Array, Object, Function etc... `of()` accepts all inputs and will `complete` as soon as all have been emitted.
+
+#### Primitive
+
+```ts
+/**
+ * output: 
+ * - next: 'hello'
+ * - complete
+ */
+of('hello').subscribe(observer);
+```
+
+#### Object/Array
+
+```ts
+/**
+ * output:
+ * - next: [1, 2, 3]
+ * - complete
+ */
+of([1, 2 ,3]).subscribe(observer);
+
+/**
+ * output:
+ * - next: {foo: 'bar'}
+ * - complete
+ */
+of({foo: 'bar'}).subscribe(observer);
+```
+
+#### Sequence of values
+
+```ts
+/**
+ * output:
+ * - next: 1
+ * - next: 2
+ * - next: 3
+ * - next: 'hello'
+ * - next: 'world'
+ * - next: {foo: 'bar'}
+ * - next: [4, 5, 6]
+ * - complete
+ */ 
+of(1, 2, 3, 'hello', 'world', { foo: 'bar' }, [4, 5, 6]).subscribe(observer);
+```
+
+`of()` when called with no arguments will `complete` immediately without any `next` notification
+
+```ts
+/**
+ * output:
+ * complete
+ */
+of().subscribe(observer);
+```
+
+## `from()`
+
+`from()`, similar to `of()`, is also a function that create an `Observable` from some value. However, the difference between `from()` and `of()` is that `from()` only accepts value that is an `Iterable`, or a `Promise`.
```

Comment:
Maybe better to word it like this: "only accepts values that are either an `Iterable` or a `Promise`"